docs/guides/configurations.md

```diff
@@ -0,0 +1,402 @@
+
+
+##### Health check servers
+
+There are two types of health check servers are built-in, liveness and readiness.
+They are used as servers for K8s liveness and readiness probe.
+
+By default, liveness servers are disabled for agent and compressor, because the liveness probes may cause to kill these components accidentally.
+
+```yaml
+agent:
+  server_config:
+    healths:
+      liveness:
+        enabled: false
+```
+
+##### Metrics servers
+
+Metrics servers are useful for debugging or monitoring Vald components.
+There are two types of metrics servers, pprof and prometheus.
+
+pprof server is a server that implemented using golang's net/http/pprof package.
+It exports profiling data that can be analyzed using [google's pprof][google-pprof] tool.
+
+prometheus server is just a [Prometheus][prometheus-io] exporter.
+To monitor Vald components by using prometheus server, it is needed to set `observability` section appropriately.
+
+#### Observability
+
+The observability features are useful for monitoring Vald components.
+They can be enabled by setting `true` at the `defaults.observability.enabled` field or override it in each components (`[component].observability.enabled`). And also, enable each features by setting `true` at the each `enabled` field.
+
+If observability features are enabled, metrics are collected every time the specified time (`observability.collector.duration`) passes.
+
+Further details are described in Observability section of [Vald operation guide][vald-operation-guide].
+
+
+### Agents
+
+#### NGT
+
+Agent-NGT uses [yahoojapan/NGT][yj-ngt] as a core library for searching vector.
+The behaviors of NGT can be configured by setting `agent.ngt` field object.
+
+The important parameters are the followings:
+
+- `agent.ngt.dimension`
+- `agent.ngt.distance_type`
+- `agent.ngt.object_type`
+
+Users should configure these parameters first for their use case.
+
+For further details, please read [NGT wiki][yj-ngt-wiki].
+
+Agent-NGT has a feature to start indexing automatically.
+The behavior of this feature can be configured with these parameters:
+
+- `agent.ngt.auto_index_duration_limit`
+- `agent.ngt.auto_index_check_duration`
+- `agent.ngt.auto_index_length`
+
+
+#### Resource requests and limits, Pod priorities
+
+Because agent places index on memory, termination of agent pods mean loss of indices.
+It is important to set resource requests and limits appropriately not to terminate agent pods.
+
+It is highly recommended to request totally 40% of cluster memory for agent pods.
+And also it is highly recommended not to set resource limits to agent pods.
+
+Pod priorities are also useful for saving agent pods from eviction.
+By default, very high priority is set to agent pods in the Chart.
+
+
+#### Pod anti-affinity
+
+It is recommended to schedule agent pods on different nodes as much as possible.
+To achieve this, the following podAntiAffinity is set by default.
```

Comment:
LGTM, but recently `topologySpreadConstraints` becomes beta in k8s 1.18 we have to use `topologySpreadConstraints` instead of `podAntiAffinity`
https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/

Reply:
Thanks. Is it okay to make another PR to add `topologySpreadConstraints`?

Reply:
I'll try to revise this part in that PR.